[Win] Fix unused parameter warnings reported by clang-cl

fujii opened the pull request:
#### 6cc3f85ca0cf2ead6c93a48cf5b0511876180a0d
<pre>
[Win] Fix unused parameter warnings reported by clang-cl
<a href="https://bugs.webkit.org/show_bug.cgi?id=262098">https://bugs.webkit.org/show_bug.cgi?id=262098</a>

Reviewed by Ross Kirsling.

clang-cl reported a lot of unused parameter warnings like the following.

&gt; Source\WebCore\platform/StaticPasteboard.cpp(143,58): error: unused parameter &apos;content&apos; [-Werror,-Wunused-parameter]

* Source/WTF/wtf/win/FileSystemWin.cpp:
(WTF::FileSystemImpl::fileID):
(WTF::FileSystemImpl::fileIDsAreEqual):
(WTF::FileSystemImpl::flushFile):
* Source/WTF/wtf/win/MemoryPressureHandlerWin.cpp:
(WTF::MemoryPressureHandler::holdOff):
* Source/WTF/wtf/win/OSAllocatorWin.cpp:
(WTF::OSAllocator::tryReserveUncommittedAligned):
* Source/WebCore/accessibility/win/AccessibilityObjectWin.cpp:
(WebCore::AccessibilityObject::detachPlatformWrapper):
* Source/WebCore/platform/graphics/Image.h:
(WebCore::Image::drawFrameMatchingSourceSize):
* Source/WebCore/platform/graphics/win/FontCacheWin.cpp:
(WebCore::metaFileEnumProc):
(WebCore::linkedFontEnumProc):
(WebCore::matchImprovingEnumProc):
(WebCore::traitsInFamilyEnumProc):
* Source/WebCore/platform/graphics/win/FontPlatformDataCairoWin.cpp:
(WebCore::FontPlatformData::platformDataInit):
* Source/WebCore/platform/graphics/win/GraphicsContextCairoWin.cpp:
(WebCore::createCairoContextWithHDC):
(WebCore::GraphicsContextCairo::GraphicsContextCairo):
* Source/WebCore/platform/graphics/win/MediaPlayerPrivateMediaFoundation.cpp:
(WebCore::MediaPlayerPrivateMediaFoundation::CustomVideoPresenter::OnClockStart):
(WebCore::MediaPlayerPrivateMediaFoundation::CustomVideoPresenter::OnClockStop):
(WebCore::MediaPlayerPrivateMediaFoundation::CustomVideoPresenter::OnClockPause):
(WebCore::MediaPlayerPrivateMediaFoundation::CustomVideoPresenter::OnClockRestart):
(WebCore::MediaPlayerPrivateMediaFoundation::CustomVideoPresenter::OnClockSetRate):
(WebCore::MediaPlayerPrivateMediaFoundation::CustomVideoPresenter::ProcessMessage):
(WebCore::MediaPlayerPrivateMediaFoundation::Direct3DPresenter::getService):
(WebCore::MediaPlayerPrivateMediaFoundation::Direct3DPresenter::presentSample):
* Source/WebCore/platform/graphics/win/MediaPlayerPrivateMediaFoundation.h:
* Source/WebCore/platform/graphics/win/SimpleFontDataWin.cpp:
(WebCore::Font::platformCreateScaledFont const):
* Source/WebCore/platform/network/win/NetworkStateNotifierWin.cpp:
(WebCore::NetworkStateNotifier::addressChangeCallback):
* Source/WebCore/platform/win/PasteboardWin.cpp:
(WebCore::Pasteboard::writePlainTextToDataObject):
* Source/WebCore/platform/win/PlatformMouseEventWin.cpp:
(WebCore::positionForEvent):
* Source/WebCore/platform/win/WidgetWin.cpp:
(WebCore::Widget::setFocus):
* Source/WebDriver/win/WebDriverServiceWin.cpp:
(WebDriver::WebDriverService::platformCompareBrowserVersions):
(WebDriver::WebDriverService::platformValidateCapability const):
* Tools/MiniBrowser/win/DialogHelper.h:
(Dialog::command):
* Tools/MiniBrowser/win/InjectedBundle.cpp:
(didReceiveMessageToPage):
* Tools/MiniBrowser/win/WebKitBrowserWindow.cpp:
(WebKitBrowserWindow::navigateToHistory):
(WebKitBrowserWindow::updateStatistics):
(WebKitBrowserWindow::decidePolicyForNavigationResponse):
(WebKitBrowserWindow::didFailProvisionalNavigation):
(WebKitBrowserWindow::didReceiveAuthenticationChallenge):
(WebKitBrowserWindow::createNewPage):
(WebKitBrowserWindow::runJavaScriptAlert):
(WebKitBrowserWindow::runJavaScriptConfirm):
(WebKitBrowserWindow::runJavaScriptPrompt):
* Tools/MiniBrowser/win/WinMain.cpp:
(wWinMain):

Canonical link: <a href="https://commits.webkit.org/268434@main">https://commits.webkit.org/268434@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/a6ec9b9944ce07c116cc53945aee58d47d112327

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/19731 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/20154 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/20771 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/21626 "Built successfully") | [  ~~🛠 wincairo~~](https://ews-build.webkit.org/#/builders/32/builds/18438 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [  ~~🧪 bindings~~](https://ews-build.webkit.org/#/builders/9/builds/19967 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/23414 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/20297 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/20006 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/19951 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/15/builds/19954 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/18/builds/17162 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/22480 "Built successfully") | 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/39/builds/17127 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/10/builds/17956 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/24235 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [  ~~🛠 🧪 jsc~~](https://ews-build.webkit.org/#/builders/20/builds/17155 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/18202 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/36/builds/18130 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/21/builds/22227 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [  ~~🛠 🧪 jsc-arm64~~](https://ews-build.webkit.org/#/builders/12/builds/19103 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 tv~~](https://ews-build.webkit.org/#/builders/7/builds/18720 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/17/builds/15864 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/23130 "Built successfully") | 
| | [  ~~🛠 tv-sim~~](https://ews-build.webkit.org/#/builders/30/builds/17882 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | | [⏳ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/JSC-ARMv7-32bits-Tests-EWS "Waiting to run tests") | 
| | [  ~~🛠 watch~~](https://ews-build.webkit.org/#/builders/4/builds/22233 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | [✅ 🛠 jsc-mips](https://ews-build.webkit.org/#/builders/24/builds/24383 "Built successfully") | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/2410 "Built successfully and passed tests") | [  ~~🛠 watch-sim~~](https://ews-build.webkit.org/#/builders/31/builds/18545 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | [✅ 🧪 jsc-mips-tests](https://ews-build.webkit.org/#/builders/3/builds/5405 "Passed tests") | 
<!--EWS-Status-Bubble-End-->

rkirsling commented on Source/WebCore/platform/graphics/win/MediaPlayerPrivateMediaFoundation.cpp:
```diff
@@ -87,7 +87,7 @@ class MediaPlayerPrivateMediaFoundation::AsyncCallback : public IMFAsyncCallback
     }
 
-    HRESULT STDMETHODCALLTYPE GetParameters(__RPC__out DWORD *pdwFlags, __RPC__out DWORD *pdwQueue) override
+    HRESULT STDMETHODCALLTYPE GetParameters(__RPC__out DWORD *, __RPC__out DWORD *) override
```
Would it be worth removing the space before the `*` in these cases?

fujii replied:
Will fix. Thank you.

rkirsling reviewed (approved):
r=me

fujii committed:
[Win] Fix unused parameter warnings reported by clang-cl
https://bugs.webkit.org/show_bug.cgi?id=262098

Reviewed by Ross Kirsling.

clang-cl reported a lot of unused parameter warnings like the following.

> Source\WebCore\platform/StaticPasteboard.cpp(143,58): error: unused parameter 'content' [-Werror,-Wunused-parameter]

* Source/WTF/wtf/win/FileSystemWin.cpp:
(WTF::FileSystemImpl::fileID):
(WTF::FileSystemImpl::fileIDsAreEqual):
(WTF::FileSystemImpl::flushFile):
* Source/WTF/wtf/win/MemoryPressureHandlerWin.cpp:
(WTF::MemoryPressureHandler::holdOff):
* Source/WTF/wtf/win/OSAllocatorWin.cpp:
(WTF::OSAllocator::tryReserveUncommittedAligned):
* Source/WebCore/accessibility/win/AccessibilityObjectWin.cpp:
(WebCore::AccessibilityObject::detachPlatformWrapper):
* Source/WebCore/platform/graphics/Image.h:
(WebCore::Image::drawFrameMatchingSourceSize):
* Source/WebCore/platform/graphics/win/FontCacheWin.cpp:
(WebCore::metaFileEnumProc):
(WebCore::linkedFontEnumProc):
(WebCore::matchImprovingEnumProc):
(WebCore::traitsInFamilyEnumProc):
* Source/WebCore/platform/graphics/win/FontPlatformDataCairoWin.cpp:
(WebCore::FontPlatformData::platformDataInit):
* Source/WebCore/platform/graphics/win/GraphicsContextCairoWin.cpp:
(WebCore::createCairoContextWithHDC):
(WebCore::GraphicsContextCairo::GraphicsContextCairo):
* Source/WebCore/platform/graphics/win/MediaPlayerPrivateMediaFoundation.cpp:
(WebCore::MediaPlayerPrivateMediaFoundation::CustomVideoPresenter::OnClockStart):
(WebCore::MediaPlayerPrivateMediaFoundation::CustomVideoPresenter::OnClockStop):
(WebCore::MediaPlayerPrivateMediaFoundation::CustomVideoPresenter::OnClockPause):
(WebCore::MediaPlayerPrivateMediaFoundation::CustomVideoPresenter::OnClockRestart):
(WebCore::MediaPlayerPrivateMediaFoundation::CustomVideoPresenter::OnClockSetRate):
(WebCore::MediaPlayerPrivateMediaFoundation::CustomVideoPresenter::ProcessMessage):
(WebCore::MediaPlayerPrivateMediaFoundation::Direct3DPresenter::getService):
(WebCore::MediaPlayerPrivateMediaFoundation::Direct3DPresenter::presentSample):
* Source/WebCore/platform/graphics/win/MediaPlayerPrivateMediaFoundation.h:
* Source/WebCore/platform/graphics/win/SimpleFontDataWin.cpp:
(WebCore::Font::platformCreateScaledFont const):
* Source/WebCore/platform/network/win/NetworkStateNotifierWin.cpp:
(WebCore::NetworkStateNotifier::addressChangeCallback):
* Source/WebCore/platform/win/PasteboardWin.cpp:
(WebCore::Pasteboard::writePlainTextToDataObject):
* Source/WebCore/platform/win/PlatformMouseEventWin.cpp:
(WebCore::positionForEvent):
* Source/WebCore/platform/win/WidgetWin.cpp:
(WebCore::Widget::setFocus):
* Source/WebDriver/win/WebDriverServiceWin.cpp:
(WebDriver::WebDriverService::platformCompareBrowserVersions):
(WebDriver::WebDriverService::platformValidateCapability const):
* Tools/MiniBrowser/win/DialogHelper.h:
(Dialog::command):
* Tools/MiniBrowser/win/InjectedBundle.cpp:
(didReceiveMessageToPage):
* Tools/MiniBrowser/win/WebKitBrowserWindow.cpp:
(WebKitBrowserWindow::navigateToHistory):
(WebKitBrowserWindow::updateStatistics):
(WebKitBrowserWindow::decidePolicyForNavigationResponse):
(WebKitBrowserWindow::didFailProvisionalNavigation):
(WebKitBrowserWindow::didReceiveAuthenticationChallenge):
(WebKitBrowserWindow::createNewPage):
(WebKitBrowserWindow::runJavaScriptAlert):
(WebKitBrowserWindow::runJavaScriptConfirm):
(WebKitBrowserWindow::runJavaScriptPrompt):
* Tools/MiniBrowser/win/WinMain.cpp:
(wWinMain):

Canonical link: https://commits.webkit.org/268434@main
